### ISSUE TITLE
Error when setting up rest connection after upgrading to 0.4.0

### ISSUE THREAD

**Issue by JerrodCarpenter** [vmware/vsphere-automation-sdk-ruby#56]
## Describe the bug

After upgrading getting an error when creating a rest connection.

* Stack trace:
```ruby
.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/vsphere-automation-runtime-0.4.1/lib/vsphere-automation-runtime/api_client.rb:132:in `merge': wrong number of arguments (given 2, expected 1) (ArgumentError)
	from .rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/vsphere-automation-runtime-0.4.1/lib/vsphere-automation-runtime/api_client.rb:132:in `add_header_params'
	from .rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/vsphere-automation-runtime-0.4.1/lib/vsphere-automation-runtime/api_client.rb:63:in `call_api'
	from .rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/vsphere-automation-cis-0.4.1/lib/vsphere-automation-cis/api/session_api.rb:59:in `create_with_http_info'
	from .rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/vsphere-automation-cis-0.4.1/lib/vsphere-automation-cis/api/session_api.rb:24:in `create'
```

## How to reproduce

```ruby
configuration = VSphereAutomation::Configuration.new.tap do |config|
  config.host = @name
  config.username = @user
  config.password = @pass
  config.scheme = 'https'
  config.verify_ssl = !@insecure
end

@rest = VSphereAutomation::ApiClient.new(configuration)
@rest.default_headers['Authorization'] = configuration.basic_auth_token
@session = VSphereAutomation::CIS::SessionApi.new(@rest)
@rest.default_headers['vmware-api-session-id'] = @session.create('').value
```

## Expected behavior

Looks like the error is raised on this last line.

## Versions

**Dev Environment:**

* Ruby: `2.4.1`
* vSphere Automation SDK: `0.4.0` & `0.4.1`
* OS: Seeing this on Mac OS and Cent OS

## Additional notes

_Add any other notes about the problem here._

**Comment by tecracer-theinen** [comment]
Just adding some context, as this has rippled into different kitchen-vcenter installations.

The difference between 0.3.0 and 0.4.0 is that a Ruby 2.6 specific merge call is used in line 132 of `api_client.rb`. Previously, that method would only accept one parameter. From Ruby 2.6 on, it can have an arbitrary number.

[Ruby 2.5.6 doc for `.merge`](https://ruby-doc.org/core-2.5.6/Hash.html#method-i-merge)
[Ruby 2.6.1 doc for `.merge`](https://ruby-doc.org/core-2.6.1/Hash.html#method-i-merge)


One obvious fix would be to have separate merge calls to restore compatibility with older Ruby versions.
```ruby
    def add_header_params(request, headers)
      header_params = @default_headers.merge(headers)
      header_params.merge!(Hash(@cookie))
      header_params.map { |name, value| request[name] = value }
    end
```

**Comment by jrgarcia** [comment]
@tecracer-theinen Yeah, I just found that out. I'll release a new version of this a little later today. Thanks!

**Comment by jrgarcia** [comment]
I just released v0.4.2. This should fix this problem. Also, just to note that authentication isn't as hard as it used to be. You don't have to setup the basic auth or session id. For an example, see the highlighted section of [this sample](https://github.com/vmware/vsphere-automation-sdk-ruby/blob/master/samples/vsphere/vcenter/vm/list-vms#L31-L44).
